pkg/network/server_test.go

```diff
@@ -126,6 +126,8 @@ func TestServerRegisterPeer(t *testing.T) {
 	const peerCount = 3
 
 	s := newTestServer(t, ServerConfig{MaxPeers: 2})
+	defer s.Shutdown()
```

Comment:
We don't need this duplicating `s.Shutdown()`, because `startWithCleanup(t, s)` calls `s.Shutdown`. If `t.Cleanup` has the same problem with zap logs as described in https://github.com/uber-go/zap/issues/687, then we need to rework all code related to `startWithCleanup(t, s)`.